### PR TITLE
[SP-8031] updated mqtt

### PR DESCRIPTION
**Pull request by eeerin**
SP-8031

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,21 +10,21 @@
...
+    "lint": "eslint --ext .js lib/** test/** examples/**",
     "test": "mocha --recursive --reporter spec test/unit 2>&1",
     "integration": "mocha --recursive --reporter spec test/integration 2>&1"
   },
   "engines": {
-    "node": ">= 8.0"
+    "node": ">= 11.0"
   },
   "dependencies": {
     "debug": ">=3 <4",
```
Comment (spacetc62) [inline package.json:21]: we can move higher on debug - was just this way cause 4 switches to es6

File: test/integration/device-spec.js [hunk 2]
```diff
@@ -7,28 +7,27 @@
...
       device = null;
-    }
-    else {
+    } else {
       done();
     }
   });
 
-  it('should connect with and without connect callback', function(done) {
+  it.skip('should connect with and without connect callback', function(done) {
```
Comment (spacetc62) [inline test/integration/device-spec.js:30]: why skip? we should get this test working

File: .eslintrc.js [hunk 3]
```diff
@@ -1,281 +1,352 @@
 module.exports = {
```
Comment (spacetc62) [inline .eslintrc.js:1]: why not just use our module directly?

File: .travis.yml [hunk 4]
```diff
@@ -5,6 +5,3 @@ node_js:
   - "13"
   - "12"
   - "11"
```
Comment (spacetc62) [inline .travis.yml:7]: remove 11, add 16

**Review by spacetc62 — CHANGES_REQUESTED** [review]

File: test/integration/device-spec.js (hunk 2 above)
Comment (eeerin) [reply test/integration/device-spec.js:30]: Oh for some when you said this one was failing on master I assumed we didnt need it anymore.

File: package.json [hunk 5]
```diff
@@ -10,21 +10,21 @@
...
-    "mqtt": "^2.18.8"
+    "mqtt": "^4.2.8"
   },
   "devDependencies": {
-    "eslint": "^6.8.0",
-    "eslint-plugin-es5": "^1.5.0",
-    "mocha": "^7.0.1",
+    "eslint": "^7.17.0",
+    "eslint-plugin-node": "^11.1.0",
+    "mocha": "^8.3.2",
```
Comment (spacetc62) [inline package.json:27]: mocha 9?

File: package.json [hunk 6]
```diff
@@ -10,21 +10,21 @@
...
   "license": "MIT",
   "scripts": {
-    "lint": "eslint *.js lib test examples",
+    "lint": "eslint --ext .js lib/** test/** examples/**",
     "test": "mocha --recursive --reporter spec test/unit 2>&1",
     "integration": "mocha --recursive --reporter spec test/integration 2>&1"
   },
   "engines": {
-    "node": ">= 8.0"
+    "node": ">= 11.0"
```
Comment (spacetc62) [inline package.json:18]: >= 12

**Review by spacetc62 — CHANGES_REQUESTED** [review]
looks like you missed a few comments from last pass, and also can you bump the .node-version file to 14.17.4?

File: .eslintrc.js (hunk 3 above)
Comment (eeerin) [reply .eslintrc.js:1]: because this is a public repo.

**Review by spacetc62 — APPROVED** [review]

**Pull request merged**